### DISCUSSION TITLE
无法导入 @formily/antd 中组件参数的类型声明

### DISCUSSION THREAD

**Discussion by Chen-Weidong** [alibaba/formily#2489] [Q&A 2.x] [answered]
我要怎么导入@formily/antd 中组件参数的类型声明呢，我看了下源码，代码没有将组件参数的类型声明导出。譬如Upload组件
![image](https://user-images.githubusercontent.com/73837770/143051423-8f1bb78b-d5b6-408b-ae00-b6ecb69dfafd.png)
这里只将组件本身导出；@formily/antd/lib/index 里也是只将所有 组件本身 导出。

**Comment by janryWang** [comment, marked as answer]
来个PR呗

**Reply by Chen-Weidong** [reply]
import React from "react";
import { Upload } from "@formily/antd";
import { message, UploadProps } from "antd";

const AvatarUpload: React.FC = (props) => {
  const uploadProps: UploadProps = {
    action: "",
    listType: "picture-card",
    beforeUpload: (file) => {
      const isImage = file.type.indexOf("image") !== -1;
      if (!isImage) {
        message.error(`${file.name} is not a image file`);
      }
      return isImage;
    },
    };
    
  return (
    <>
      <Upload {...props} {...uploadProps}>
        <div style={{ marginTop: 8 }}>Upload</div>
      </Upload>
    </>
  );
};
export default AvatarUpload;

上面这个是demo；然后可以看下面这个图，@formily/antd 里的 Upload 的组件参数类型跟 antd里的不一样，同时@formily/antd也没有将组件参数类型导出，所以在ts中像 ref 等需要提供类型的地方会遇到问题。
![image](https://user-images.githubusercontent.com/73837770/143063419-c26d5169-0bb1-4206-8c07-4013f2f517fd.png)

我尝试进行修改，把源码中Upload组件的参数类型导出来，在代码中使用它，就没报错了。所以我想，是不是应该把@formily/antd中组件的参数类型都给导出来。
![image](https://user-images.githubusercontent.com/73837770/143069157-d78391ad-8a65-49cf-9b62-70fd45d4581d.png)
![image](https://user-images.githubusercontent.com/73837770/143069294-4e17dfda-300e-4fa3-955c-f998a6f95174.png)

**Reply by janryWang** [reply]
嗯嗯，应该是需要的，遗漏掉了
